std/math/complex/sqrt.zig

```diff
@@ -0,0 +1,140 @@
+const std = @import("../../index.zig");
+const debug = std.debug;
+const math = std.math;
+const cmath = math.complex;
+const Complex = cmath.Complex;
+
+// NOTE: Returning @typeOf(z) here causes issues when trying to access the value. This is
```

Comment:
Was there a current issue regarding this, cannot recall?

Reply:
Looks related to #733?

Reply:
Is there an issue open for this? I'm unaware of this problem.

Reply:
Are you referring to the fact that if you pass a struct to a parameter that accepts `var` then it ends up being a `&const S` instead of `S`?

I don't think that will affect calling with `c.re` vs calling with `re`. I'll run the tests locally to make sure.

Reply:
Here is an example. Release mode actually crashes with a SIGSEGV here. May not be 100% the same issue.

```
const std = @import("std");

fn Complex(comptime T: type) type {
    return struct {
        const Self = this;

        re: T,
        im: T,

        pub fn new(re: T, im: T) Self {
            return Self {
                .re = re,
                .im = im,
            };
        }
    };
}

fn clone1(z: var) @typeOf(z) {
    const T = @typeOf(z.re);
    return Complex(T).new(z.im, z.im);
}

fn clone2(z: var) Complex(@typeOf(z.re)) {
    const T = @typeOf(z.re);
    return Complex(T).new(z.im, z.im);
}

pub fn main() void {
    const s = Complex(f32).new(5.4, 3.3);

    const a1 = clone1(s);
    const a2 = clone2(s);

    std.debug.warn("{} {}\n", a1.re, a1.im);
    std.debug.warn("{} {}\n", a2.re, a2.im);
}
```

Reply:
Maybe more similar is if you use something like the following behaviour which I believe I was using at one point.

```
fn clone1(z: var) @typeOf(z) {
    const T = @typeOf(z.re);
    return Complex(T).new(z.im, math.ln(z.re));
}

fn clone2(z: var) Complex(@typeOf(z.re)) {
    const s = clone1(z);
    return *s;
}
```

Which gives the following debug output:

```
-1.43565942e-38 4.59135441e-41
3.29999995 1.68639898
```